core/templates/dev/head/domain/subtopic_viewer/SubtopicDataObjectFactory.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating instances of frontend
+ * subtopic data domain objects.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { SubtitledHtmlObjectFactory } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { SubtopicPageContentsObjectFactory } from
+  'domain/topic/SubtopicPageContentsObjectFactory';
+import { RecordedVoiceoversObjectFactory } from
+  'domain/exploration/RecordedVoiceoversObjectFactory';
+import { VoiceoverObjectFactory } from
+  'domain/exploration/VoiceoverObjectFactory';
+
+export class SubtopicData {
```

Comment:
@DubeySandeep  I named this class `SubtopicData` because the variables used for representing this before were named like this.

Also, There is a SubtopicObjectFactory already 

 https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/topic/SubtopicObjectFactory.ts#L26-L35

Reply:
> Also, There is a SubtopicObjectFactory already

> https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/topic/SubtopicObjectFactory.ts#L26-L35

@nishantwrp, This is why I was confused with the name as we already have a Subtopic class. I would suggest you look into the usability of this class and rename it accordingly. ([If required] I can help you if you can provide info on what will be the use of this data.)

Reply:
@DubeySandeep SubtopicData is the data that needs to be displayed to a learner on the subtopic page.

Reply:
> SubtopicData is the data that needs to be displayed to a learner on the subtopic page.

Thanks, still I'm not sure how is this different from `SubtopicObject`, can you please check?

Reply:
@DubeySandeep I can't find the exact use of SubtopicObject except it is used in TopicObjectFactory

Reply:
@aks681, Can you please us with info on what's the use of this data and can you suggest a good name for the class? 

(If it's just representing the subtopic in frontend then maybe: `RepresentableSubtopic` )

Reply:
@aks681 We already have `SubtopicPageObjectFactory` [here](https://github.com/oppia/oppia/blob/develop/core/templates/dev/head/domain/topic/SubtopicPageObjectFactory.ts#L35), is that the same?

Reply:
Yeah, conceptually, both are the same objects. The existing one is for the editors. I'm fine with reusing that here, though then, some fields in it would become undefined in this use case.
I think it's fine to create this new factory, with name as something like `SubtopicPageDataObjectFactory` or `ReadOnlySubtopicPageObjectFactory`. Similar to how `StoryNodeObjectFactory` and `ReadOnlyStoryNodeObjectFactory` exists.

Reply:
Yeah, let's make it `ReadOnlySubtopicPageObjectFactory`! Thanks for the help @aks681!

Reply:
@DubeySandeep What do we name the class `SubtopicData`

Reply:
You can rename that to `SubtopicPageData` or `ReadOnlySubtopicPageData`.

Reply:
I would suggest making it ReadOnlySubtopicPageData to avoid confusion and having a consistent structure.